cosmpy/aerial/client/__init__.py

```diff
@@ -0,0 +1,281 @@
+    QueryTimeoutError,
+)
+from cosmpy.aerial.tx import SigningCfg, Transaction
+from cosmpy.aerial.tx_helpers import MessageLog, SubmittedTx, TxResponse
+from cosmpy.aerial.urls import Protocol, parse_url
+from cosmpy.aerial.wallet import Wallet
+from cosmpy.auth.rest_client import AuthRestClient
+from cosmpy.bank.rest_client import BankRestClient
+from cosmpy.common.rest_client import RestClient
+from cosmpy.cosmwasm.rest_client import CosmWasmRestClient
+from cosmpy.crypto.address import Address
+from cosmpy.protos.cosmos.auth.v1beta1.auth_pb2 import BaseAccount
+from cosmpy.protos.cosmos.auth.v1beta1.query_pb2 import QueryAccountRequest
+from cosmpy.protos.cosmos.auth.v1beta1.query_pb2_grpc import QueryStub as AuthGrpcClient
+from cosmpy.protos.cosmos.bank.v1beta1.query_pb2 import QueryBalanceRequest
+from cosmpy.protos.cosmos.bank.v1beta1.query_pb2_grpc import QueryStub as BankGrpcClient
+from cosmpy.protos.cosmos.staking.v1beta1.query_pb2_grpc import (
+    QueryStub as StakingGrpcClient,
+)
+from cosmpy.protos.cosmos.tx.v1beta1.service_pb2 import (
+    BroadcastMode,
+    BroadcastTxRequest,
+    GetTxRequest,
+)
+from cosmpy.protos.cosmos.tx.v1beta1.service_pb2_grpc import ServiceStub as TxGrpcClient
+from cosmpy.protos.cosmwasm.wasm.v1.query_pb2_grpc import (
+    QueryStub as CosmWasmGrpcClient,
+)
+from cosmpy.staking.rest_client import StakingRestClient
+from cosmpy.tx.rest_client import TxRestClient
+
+DEFAULT_QUERY_TIMEOUT_SECS = 15
+DEFAULT_QUERY_INTERVAL_SECS = 2
+
+
+@dataclass
+class Account:
+    address: Address
+    number: int
+    sequence: int
+
+
+class LedgerClient:
+    def __init__(self, cfg: NetworkConfig):
+        cfg.validate()
+
+        self._network_config = cfg
+
+        parsed_url = parse_url(cfg.url)
+
+        if parsed_url.protocol == Protocol.GRPC:
+            if parsed_url.secure:
+                with open(certifi.where(), "rb") as f:
+                    trusted_certs = f.read()
+                credentials = grpc.ssl_channel_credentials(
+                    root_certificates=trusted_certs
+                )
+                grpc_client = grpc.secure_channel(parsed_url.host_and_port, credentials)
+            else:
+                grpc_client = grpc.insecure_channel(parsed_url.host_and_port)
+
+            self.wasm = CosmWasmGrpcClient(grpc_client)
+            self.auth = AuthGrpcClient(grpc_client)
+            self.txs = TxGrpcClient(grpc_client)
+            self.bank = BankGrpcClient(grpc_client)
+            self.staking = StakingGrpcClient(grpc_client)
+        else:
+            rest_client = RestClient(parsed_url.rest_url)
+
+            self.wasm = CosmWasmRestClient(rest_client)  # type: ignore
+            self.auth = AuthRestClient(rest_client)  # type: ignore
+            self.txs = TxRestClient(rest_client)  # type: ignore
+            self.bank = BankRestClient(rest_client)  # type: ignore
+            self.staking = StakingRestClient(rest_client)  # type: ignore
```

Comment:
Just note: We will have to refactor these clients. because each GRPC module has multiple clients - QueryStub, ServiceStub, MsgStub. 

So in the case of StakingRestClient, it is just QueryStub functionality, but there are also others.

Reply:
👍  - I will do this as part of #105